File: wear_os_app/app/src/main/java/com/pathofnur/watch/domain/WatchModels.kt
package com.pathofnur.watch.domain

import java.time.Instant

enum class PrayerState { Pending, Completed }
enum class SourceType { Watch, Phone, Synced }
enum class DhikrMode { Preset33, Preset99, Free }
enum class DhikrInteractionMode { ManualTap, SilentTap, GuidedPulse }
enum class SyncItemType { PrayerComplete, PrayerUncomplete, DhikrIncrement, DhikrReset, SessionComplete, QuranPlaybackCommand }
enum class SyncStatus { Pending, Synced, Failed }

data class WatchDaySnapshot(
    val date: String,
    val timezone: String,
    val nextPrayerId: String,
    val nextPrayerTimeLabel: String,
    val nextPrayerCountdownLabel: String?,
    val currentPrayerId: String?,
    val completedPrayerCount: Int,
    val totalPrayerCount: Int,
    val dhikrTodayCount: Int,
    val xpToday: Int,
    val oceanDropsToday: Int,
    val streakDays: Int,
    val lastSyncLabel: String?
)

data class WatchPrayerStatus(
    val prayerId: String,
    val displayName: String,
    val scheduledTimeLabel: String,
    val status: PrayerState,
    val completedAt: Instant?,
    val source: SourceType
)

data class WatchDhikrSession(
    val sessionId: String,
    val mode: DhikrMode,
    val targetCount: Int?,
    val currentCount: Int,
    val startedAt: Instant,
    val updatedAt: Instant,
    val completed: Boolean,
    val completionMilestoneReached: Boolean,
    val rewardEligible: Boolean,
    val rewardGranted: Boolean,
    val interactionMode: DhikrInteractionMode,
    val pulseIntervalSeconds: Double?,
    val isPulseRunning: Boolean,
    val pulseTargetCount: Int?,
    val pulseCurrentStep: Int,
    val paceReminderShownAt: Instant?,
    val pausedAt: Instant?,
    val source: SourceType
)

enum class QuranPlaybackSourceType { Phone, Watch }
enum class QuranPlaybackCommand {
    Play,
    Pause,
    Next,
    Previous,
    SeekForward15,
    SeekBack15,
    SwitchOutputPhone,
    SwitchOutputWatch,
    ResumeLast
}

data class QuranPlaybackSnapshot(
    val playbackSessionId: String,
    val sourceType: QuranPlaybackSourceType,
    val isPlaying: Boolean,
    val surahId: Int?,
    val surahName: String,
    val reciterId: String,
    val reciterName: String,
    val currentAyah: Int?,
    val positionSeconds: Int,
    val durationSeconds: Int?,
    val isBuffering: Boolean,
    val lastUpdatedAt: Instant
)

data class WatchAudioAvailabilitySnapshot(
    val watchPlaybackAvailable: Boolean,
    val downloadedSurahIds: List<Int>,
    val recentPlayableItems: List<String>,
    val lastSyncedAt: Instant?
)

data class WatchSyncQueueItem(
    val id: String,
    val type: SyncItemType,
    val payload: Map<String, String>,
    val createdAt: Instant,
    val syncStatus: SyncStatus,
    val retryCount: Int
)

enum class WatchNotificationKind { Initial, FollowUp, Snooze, Dhikr }
enum class WatchReminderStatus { Scheduled, Fired, Cancelled, Expired }
enum class WatchNotificationActionType { MarkPrayed, Snooze, OpenApp, OpenDhikr, Dismiss }

data class WatchNotificationSettings(
    val prayerNotificationsEnabled: Boolean,
    val enabledPrayerIds: List<String>,
    val followUpReminderEnabled: Boolean,
    val followUpDelayMinutes: Int,
    val snoozeDurationMinutes: Int,
    val quietModeEnabled: Boolean,
    val dhikrReminderEnabled: Boolean
) {
    companion object {
        val Default = WatchNotificationSettings(
            prayerNotificationsEnabled = true,
            enabledPrayerIds = listOf("fajr", "dhuhr", "asr", "maghrib", "isha"),
            followUpReminderEnabled = true,
            followUpDelayMinutes = 20,
            snoozeDurationMinutes = 10,
            quietModeEnabled = false,
            dhikrReminderEnabled = false
        )
    }
}

data class ScheduledPrayerReminder(
    val id: String,
    val prayerId: String,
    val date: String,
    val kind: WatchNotificationKind,
    val scheduledAt: Instant,
    val status: WatchReminderStatus,
    val dedupKey: String
)

data class NotificationActionLog(
    val id: String,
    val prayerId: String?,
    val actionType: WatchNotificationActionType,
    val timestamp: Instant,
    val source: String
)

data class NotificationDedupState(
    val dedupKey: String,
    val lastScheduledAt: Instant?,
    val lastFiredAt: Instant?,
    val lastCancelledAt: Instant?
)

enum class DhikrMilestone { None, Milestone33, Milestone66, Milestone99, TargetCompleted }

data class DhikrIncrementOutcome(
    val session: WatchDhikrSession,
    val todayTotal: Int,
    val milestone: DhikrMilestone,
    val rewardGranted: Boolean,
    val shouldQueueCompletion: Boolean
)

File: apple_watch_app/PathOfNurWatch/Domain/WatchModels.swift
import Foundation

enum WatchPrayerState: String, Codable {
    case pending
    case completed
}

enum WatchSource: String, Codable {
    case phone
    case watch
    case synced
}

enum WatchDhikrMode: String, Codable {
    case preset33
    case preset99
    case free
}

enum WatchDhikrInteractionMode: String, Codable {
    case manualTap
    case silentTap
    case guidedPulse
}

enum WatchSyncItemType: String, Codable {
    case prayerComplete
    case prayerUncomplete
    case dhikrIncrement
    case dhikrReset
    case sessionComplete
    case quranPlaybackCommand
}

enum WatchSyncStatus: String, Codable {
    case pending
    case synced
    case failed
}

struct WatchDaySnapshot: Codable {
    var date: Date
    var timezone: String
    var nextPrayerId: String
    var nextPrayerTime: Date
    var currentPrayerId: String?
    var completedPrayerCount: Int
    var totalPrayerCount: Int
    var dhikrTodayCount: Int
    var xpToday: Int
    var oceanDropsToday: Int
    var streakDays: Int
    var lastSyncAt: Date?
}

struct WatchPrayerStatus: Identifiable, Codable {
    var id: String { prayerId }
    var prayerId: String
    var displayName: String
    var scheduledTime: Date
    var status: WatchPrayerState
    var completedAt: Date?
    var source: WatchSource
}

struct WatchDhikrSession: Codable {
    var sessionId: String
    var mode: WatchDhikrMode
    var targetCount: Int?
    var currentCount: Int
    var startedAt: Date
    var updatedAt: Date
    var completed: Bool
    var completionMilestoneReached: Bool
    var rewardEligible: Bool
    var rewardGranted: Bool
    var interactionMode: WatchDhikrInteractionMode
    var pulseIntervalSeconds: Double?
    var isPulseRunning: Bool
    var pulseTargetCount: Int?
    var pulseCurrentStep: Int
    var paceReminderShownAt: Date?
    var pausedAt: Date?
    var source: WatchSource
}

enum WatchQuranPlaybackSourceType: String, Codable {
    case phone
    case watch
}

enum WatchQuranPlaybackCommand: String, Codable {
    case play
    case pause
    case next
    case previous
    case seekForward15
    case seekBack15
    case switchOutputPhone
    case switchOutputWatch
    case resumeLast
}

struct WatchQuranPlaybackSnapshot: Codable {
    var playbackSessionId: String
    var sourceType: WatchQuranPlaybackSourceType
    var isPlaying: Bool
    var surahId: Int?
    var surahName: String
    var reciterId: String
    var reciterName: String
    var currentAyah: Int?
    var positionSeconds: Int
    var durationSeconds: Int?
    var isBuffering: Bool
    var lastUpdatedAt: Date
}

struct WatchAudioAvailabilitySnapshot: Codable {
    var watchPlaybackAvailable: Bool
    var downloadedSurahIds: [Int]
    var recentPlayableItems: [String]
    var lastSyncedAt: Date?
}

struct WatchSyncQueueItem: Identifiable, Codable {
    var id: String
    var type: WatchSyncItemType
    var payload: [String: String]
    var createdAt: Date
    var syncStatus: WatchSyncStatus
    var retryCount: Int
}

enum WatchNotificationKind: String, Codable {
    case initial
    case followUp
    case snooze
    case dhikr
}

enum WatchReminderStatus: String, Codable {
    case scheduled
    case fired
    case cancelled
    case expired
}

enum WatchNotificationActionType: String, Codable {
    case markPrayed
    case snooze
    case openApp
    case openDhikr
    case dismiss
}

struct WatchNotificationSettings: Codable {
    var prayerNotificationsEnabled: Bool
    var enabledPrayerIds: [String]
    var followUpReminderEnabled: Bool
    var followUpDelayMinutes: Int
    var snoozeDurationMinutes: Int
    var quietModeEnabled: Bool
    var dhikrReminderEnabled: Bool

    static let `default` = WatchNotificationSettings(
        prayerNotificationsEnabled: true,
        enabledPrayerIds: ["fajr", "dhuhr", "asr", "maghrib", "isha"],
        followUpReminderEnabled: true,
        followUpDelayMinutes: 20,
        snoozeDurationMinutes: 10,
        quietModeEnabled: false,
        dhikrReminderEnabled: false
    )
}

struct ScheduledPrayerReminder: Identifiable, Codable {
    var id: String
    var prayerId: String
    var date: Date
    var kind: WatchNotificationKind
    var scheduledAt: Date
    var status: WatchReminderStatus
    var dedupKey: String
}

struct NotificationActionLog: Identifiable, Codable {
    var id: String
    var prayerId: String?
    var actionType: WatchNotificationActionType
    var timestamp: Date
    var source: String
}

struct NotificationDedupState: Codable {
    var dedupKey: String
    var lastScheduledAt: Date?
    var lastFiredAt: Date?
    var lastCancelledAt: Date?
}

enum DhikrMilestone: String, Codable {
    case none
    case milestone33
    case milestone66
    case milestone99
    case targetCompleted
}

struct DhikrIncrementOutcome {
    var session: WatchDhikrSession
    var todayTotal: Int
    var milestone: DhikrMilestone
    var rewardGranted: Bool
    var shouldQueueCompletion: Bool
}

enum SnapshotFreshnessStatus: String, Codable {
    case fresh
    case aging
    case stale
    case expired
}

struct SnapshotFreshnessState: Codable {
    var snapshotDate: Date?
    var lastRefreshAt: Date?
    var lastSuccessfulSyncAt: Date?
    var freshnessStatus: SnapshotFreshnessStatus
    var reason: String
}

struct PrayerPhaseState: Codable {
    var currentPrayerId: String?
    var nextPrayerId: String?
    var nextPrayerTime: Date?
    var lastEvaluatedAt: Date
}

enum DayRolloverStatus: String, Codable {
    case idle
    case needed
    case processing
    case complete
    case failed
}

struct DayRolloverState: Codable {
    var activeDate: Date
    var lastRolloverCheckAt: Date?
    var lastRolloverCompletedAt: Date?
    var rolloverStatus: DayRolloverStatus
}

enum BackgroundRefreshEventType: String, Codable {
    case appLaunch
    case appResume
    case scheduledRefresh
    case nextPrayerTransition
    case dayRollover
    case syncComplete
    case manualRefresh
    case prayerCompletion
    case dhikrUpdate
    case notificationAction
}

struct BackgroundRefreshEvent: Identifiable, Codable {
    var id: String
    var type: BackgroundRefreshEventType
    var timestamp: Date
    var handled: Bool
}

struct RefreshResult {
    var snapshotUpdated: Bool
    var notificationsRescheduled: Bool
    var tilesRefreshed: Bool
    var complicationsRefreshed: Bool
    var syncAttempted: Bool
    var rolloverPerformed: Bool
    var errors: [String]
}
